report_test.go

```diff
@@ -1,109 +1,104 @@
-    "testing"
+	"bytes"
+	"errors"
+	"net"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
 )
 
 var (
-    testPort string
-    fakeURL string
+	testPort = ":5123"
+	fakeURL  = "http://127.0.0.1" + testPort
 )
 
-func FakeServer(finishServer <-chan bool, startServer chan<- bool) {
-    ts := httptest.NewUnstartedServer(
-        http.FileServer(http.Dir("./sample-html")))
-    l, _ := net.Listen("tcp", testPort)
-    ts.Listener = l
-    ts.Start()
-    startServer<-true
-    <-finishServer
-    defer ts.Close()
-}
-
 func TestDownloadURL(t *testing.T) {
-    endFakeServer := make(chan bool)
-    startFakeServer := make(chan bool)
-    go FakeServer(endFakeServer, startFakeServer)
-    <-startFakeServer
-    DownloadURL(fakeURL + "/index.html", "a.html")
-    defer os.Remove("a.html")
-    endFakeServer<-true
+	handler := http.FileServer(http.Dir("./testdata"))
+	ts := httptest.NewServer(handler)
+	l, _ := net.Listen("tcp", testPort)
+	ts.Listener = l
+	defer ts.Close()
+
+	err := DownloadURL(fakeURL+"/index.html", "a.html")
+	if err != nil {
+		t.Fatal(err)
+	}
+	//defer os.Remove("a.html")
 
-    // Check downloaded html
-    html1, err := ioutil.ReadFile("a.html")
-    if err != nil {
-        t.Fatalf("%v\n", err)
-    }
-    html2, err := ioutil.ReadFile("sample-html/index.html")
-    if err != nil {
-        t.Fatalf("%v\n", err)
-    }
-    if !bytes.Equal(html1, html2) {
-        t.Fatalf("%v\n", "Downloaded html and server html are not equal")
-    }
+	// Check downloaded html
+	html1, err := ioutil.ReadFile("a.html")
+	if err != nil {
+		t.Fatal(err)
+	}
+	html2, err := ioutil.ReadFile("testdata/index.html")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(html1, html2) {
+		t.Fatalf("%v\n", "Downloaded html and server html are not equal")
+	}
 }
 
 func TestDownloadMoss(t *testing.T) {
-    endFakeServer := make(chan bool)
-    startFakeServer := make(chan bool)
-    go FakeServer(endFakeServer, startFakeServer)
-    <-startFakeServer
-    DownloadMoss(fakeURL)
-    defer os.RemoveAll("moss.stanford.edu")
-    endFakeServer<-true
+	handler := http.FileServer(http.Dir("./testdata"))
+	ts := httptest.NewServer(handler)
+	l, _ := net.Listen("tcp", testPort)
+	ts.Listener = l
+	defer ts.Close()
 
-    // Check downloaded html
-    err := filepath.Walk("moss.stanford.edu", func(path string, info os.FileInfo, err error) error {
-        if info.IsDir() {
-            return nil
-        }
-        html1, err := ioutil.ReadFile("moss.stanford.edu/" + filepath.Base(path))
-        if err != nil {
-            return err
-        }
-        html2, err := ioutil.ReadFile("sample-html/" + filepath.Base(path))
-        if err != nil {
-            return err
-        }
-        if !bytes.Equal(html1, html2) {
-            return errors.New(filepath.Base(path) + " are not equal")
-        }
-        return nil
-    })
-    if err != nil {
-        t.Fatalf("%v\n", err)
-    }
+	err := Download(fakeURL)
```

Comment:
why do you need to make a custom `Listener`? Can't you use `ts.URL` instead of `fakeURL` here?